lib/AsyncStorage.js

```diff
@@ -21,6 +20,21 @@ const RCTAsyncStorage =
   NativeModules.RNC_AsyncSQLiteDBStorage ||
   NativeModules.AsyncLocalStorage;
 
+type ReadOnlyArrayString = $ReadOnlyArray<string>;
```

Comment:
What about exporting this to different file, say `types/flow.js` and then include it in `.flowconfig`?

Reply:
@Krizzu 
There is some troubles when using `.flowconfig`. For example, hyperlink(Go to definition) in vscode does not work properly. 

I think the approach below looks ok.

- Added `AsyncStorageType.js` to define types and separated it from `AsyncStorage.js`.
- Added `AsyncStorageType.js.flow` to provide more clean autocomplete.

Let me know what you think about this.

Reply:
On the second thought, having types defined in the same file is better than have 2 different files with types. Thought extracting flow types into another file would be better supported by VScode.

If you would revert 1 commit back, I'd be more than happy to merge it 👍 
Sorry for confusion!

Reply:
@Krizzu Sure. I had a similar idea. I modified it as you said.

Reply:
Great, thank You.
In that case, where we have Types in the same file, do we still need `.js.flow` file? Any advantages to having it?
Also, can you rebase with `master`, please?

Reply:
@Krizzu With `AsyncStorage.js.flow`, we can use more organized auto-completion in VSCode, but I realized there was a problem with the function to jump directly to `AsyncStorage.js` to check the code.

It seemed to have more disadvantages than merits, so I decided to delete `AsyncStorage.js.flow`. Sorry for confusion!

I finished what you said.